connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -763,6 +763,9 @@ private void onPartitionsRemoved(Collection<TopicPartition> partitions, boolean
                 return;
 
             try {
+                for (TopicPartition partition: partitions) {
+                    origOffsets.remove(partition);
+                }
```

Comment:
Any reason not to put this in `closePartitions`, where we also clear out entries from `currentOffsets`?
Also, this can be simplified:
```java
origOffsets.keySet().removeAll(partitions);
```

Reply:
Makes sense